hook-janitor/src/main.rs

```diff
@@ -0,0 +1,86 @@
+use axum::Router;
+use cleanup::{Cleaner, CleanerModeName};
+use config::Config;
+use envconfig::Envconfig;
+use eyre::Result;
+use futures::future::{select, Either};
+use kafka_producer::create_kafka_producer;
+use std::{str::FromStr, time::Duration};
+use tokio::sync::Semaphore;
+use webhooks::WebhookCleaner;
+
+use hook_common::metrics;
+
+mod cleanup;
+mod config;
+mod handlers;
+mod kafka_producer;
+mod webhooks;
+
+async fn listen(app: Router, bind: String) -> Result<()> {
+    let listener = tokio::net::TcpListener::bind(bind).await?;
+
+    axum::serve(listener, app).await?;
+
+    Ok(())
+}
+
+async fn cleanup_loop(cleaner: Box<dyn Cleaner>, interval_secs: u64) -> Result<()> {
```

Comment:
nit: I'd avoid returning a result if there's nothing here that can return an error, otherwise we force callers to handle an error that will never happen. But it may be that this signature is to accommodate `cleanup` returning errors in the future?

Reply:
Nope, I think this was from an earlier iteration where I was having trouble making the `select` happy, but I can remove this now for sure.